src/bots/fillerLite.ts

```diff
@@ -0,0 +1,1274 @@
+		triggerableNodes: Array<NodeToTrigger>
+	): {
+		filteredFillableNodes: Array<NodeToFill>;
+		filteredTriggerableNodes: Array<NodeToTrigger>;
+	} {
+		const seenFillableNodes = new Set<string>();
+		const filteredFillableNodes = fillableNodes.filter((node) => {
+			const sig = getNodeToFillSignature(node);
+			if (seenFillableNodes.has(sig)) {
+				return false;
+			}
+			seenFillableNodes.add(sig);
+			return this.filterFillableNodes(node);
+		});
+
+		const seenTriggerableNodes = new Set<string>();
+		const filteredTriggerableNodes = triggerableNodes.filter((node) => {
+			const sig = getNodeToTriggerSignature(node);
+			if (seenTriggerableNodes.has(sig)) {
+				return false;
+			}
+			seenTriggerableNodes.add(sig);
+			return this.filterTriggerableNodes(node);
+		});
+
+		return {
+			filteredFillableNodes,
+			filteredTriggerableNodes,
+		};
+	}
+
+	protected async executeTriggerablePerpNodesForMarket(
+		triggerableNodes: Array<NodeToTrigger>
+	) {
+		for (const nodeToTrigger of triggerableNodes) {
+			nodeToTrigger.node.haveTrigger = true;
+			logger.info(
+				`trying to trigger (account: ${nodeToTrigger.node.userAccount.toString()}) order ${nodeToTrigger.node.order.orderId.toString()}`
+			);
+
+			const userAccount = this.orderSubscriber.usersAccounts.get(
+				nodeToTrigger.node.userAccount.toString()
+			).userAccount;
+
+			const nodeSignature = getNodeToTriggerSignature(nodeToTrigger);
+			this.triggeringNodes.set(nodeSignature, Date.now());
+
+			this.driftClient
+				.triggerOrder(
+					nodeToTrigger.node.userAccount,
+					userAccount,
+					nodeToTrigger.node.order
+				)
+				.then((txSig) => {
+					logger.info(
+						`Triggered user (account: ${nodeToTrigger.node.userAccount.toString()}) order: ${nodeToTrigger.node.order.orderId.toString()}`
+					);
+					logger.info(`Tx: ${txSig}`);
+				})
+				.catch((error) => {
+					nodeToTrigger.node.haveTrigger = false;
+
+					const errorCode = getErrorCode(error);
+					if (
+						!errorCodesToSuppress.includes(errorCode) &&
+						!(error as Error).message.includes('Transaction was not confirmed')
+					) {
+						logger.error(
+							`Error (${errorCode}) triggering order for user (account: ${nodeToTrigger.node.userAccount.toString()}) order: ${nodeToTrigger.node.order.orderId.toString()}`
+						);
+						logger.error(error);
+						webhookMessage(
+							`[${
+								this.name
+							}]: :x: Error (${errorCode}) triggering order for user (account: ${nodeToTrigger.node.userAccount.toString()}) order: ${nodeToTrigger.node.order.orderId.toString()}\n${
+								error.stack ? error.stack : error.message
+							}`
+						);
+					}
+				})
+				.finally(() => {
+					this.removeTriggeringNodes(nodeToTrigger);
+				});
+		}
+
+		const user = this.driftClient.getUser();
+		this.attemptedTriggersCounter.add(
+			triggerableNodes.length,
+			metricAttrFromUserAccount(
+				user.userAccountPublicKey,
+				user.getUserAccount()
+			)
+		);
+	}
+
+	protected async tryFill() {
```

Comment:
looks like this guy can be largely using the parent too! just need to refactor to allow the option to use `ordersubscriber`